### ISSUE TITLE
Controllers with Controller in the name

### ISSUE THREAD

**Issue by smogHead** [T4MVC/T4MVC#27]
If you have a Controller called HelpControllerController because your domain modem is called HelpController T4MVC blows on generation. 

I have fixed it in my version and changed as follows: 

```
public string Name
{
    get
    {
        // remove controller suffix (if it exists)
        return ClassName.Replace(ControllerSuffix, "");
   }
}

TO 

public string Name
{
    get
    {
        // remove controller suffix (if it exists)
 //       return ClassName.Replace(ControllerSuffix, "");
        return ClassName.Substring(0, ClassName.Length - ControllerSuffix.Length);
    }
}
```

Hope it helps someone one day

Steve

**Comment by davidebbo** [comment]
You'd need to check that it has the suffix at all before stripping it. But other than that, it would be a good fix to make.

**Comment by davidebbo** [comment]
Ok, fix is in 3.15.3.
